ql/src/java/org/apache/hadoop/hive/ql/ddl/table/create/like/CreateTableLikeOperation.java

```diff
@@ -155,6 +158,15 @@ private void setTableParameters(Table tbl, Map<String, String> originalPropertie
     if (desc.getTblProps() != null) {
       tbl.setParameters(desc.getTblProps());
     }
+    String paramsStr = HiveConf.getVar(context.getConf(), HiveConf.ConfVars.DDL_CTL_PARAMETERS_WHITELIST);
```

Comment:
nit: can we move this changes above 
```
  if (desc.getTblProps() != null) {
      tbl.setParameters(desc.getTblProps());
    }
```
and change it to:
```
 String paramsStr = HiveConf.getVar(context.getConf(), HiveConf.ConfVars.DDL_CTL_PARAMETERS_WHITELIST);
    if (paramsStr != null) {
      Set<String> retainer = new HashSet<String>(Arrays.asList(paramsStr.split(",")));
      originalProperties.entrySet().stream().filter(entry -> retainer.contains(entry.getKey()))
          .forEach(entry -> tbl.getParameters().put(entry.getKey(), entry.getValue()));
    }

Reply:
Will the issue arise when re-adding the whitelisted original properties from `originalProperties` back to the table, even if they've been explicitly overridden?

Reply:
you can make this change after:
```
if (desc.getTblProps() != null) {
      tbl.setParameters(desc.getTblProps());
    }
    ```
so it will be overwritten if some are specified explicitly